core/src/main/scala/scala/pickling/Pickler.scala

```diff
@@ -101,16 +102,43 @@ object PicklerUnpickler {
   }
 }
 
-/** A pickler which will register itself with the runtime pickler registery. */
-abstract class AutoRegister[T: FastTypeTag](name: String) extends AbstractPicklerUnpickler[T] {
-  val tag = implicitly[FastTypeTag[T]]
+object AutoRegister {
+  import scala.pickling.internal.currentRuntime
+
+  private def isLookupEnabled: Boolean =
+    !currentRuntime.isInstanceOf[NoReflectionRuntime]
```

Comment:
nit: Maybe this should be a flag on `PicklerRegistry` in case users create their own....